### ISSUE TITLE
Something is wrong with the new update using the Windows binaries 

### ISSUE THREAD

**Issue by hellishvictor** [TomWright/dasel#390]
**Describe the bug**
Hi, I've downloaded from the release section the new v2.6.0 Windows executables, and using "--help", it shows:

**dasel_windows_386 --help**

```
Exception 0xc0000005 0x8 0x0 0x0
PC=0x0

runtime.asmstdcall(0x20)
        /opt/hostedtoolcache/go/1.22.0/x64/src/runtime/sys_windows_386.s:44 +0x2a fp=0x18fe28 sp=0x1
8fe24 pc=0x46f1da
eax     0x8
ebx     0xcdabd0
ecx     0x0
edx     0x18fe58
edi     0x18fe2c
esi     0x18fe8c
ebp     0x18fe2c
esp     0x18fe20
eip     0x0
eflags  0x10206
cs      0x23
fs      0x53
gs      0x2b
```

**dasel_windows_amd64 --help**
```
Exception 0xc0000005 0x8 0x0 0x0
PC=0x0

runtime.asmstdcall(0x22fcf8)
        /opt/hostedtoolcache/go/1.22.0/x64/src/runtime/sys_windows_amd64.s:75 +0x7a fp=0x22fc80 sp=0
x22fc60 pc=0x46f4fa
rax     0x0
rbx     0xd609f8
rcx     0xde5aa8
rdx     0x20
rdi     0x7fffffde000
rsi     0x22fe78
rbp     0x22fdc0
rsp     0x22fc58
r8      0x0
r9      0x22fee8
r10     0x2fa2e8
r11     0xc000004000
r12     0x22fec8
r13     0x0
r14     0xd5f640
r15     0x3
rip     0x0
rflags  0x10293
cs      0x33
fs      0x53
gs      0x2b
```
I switched back to the previous working version v2.5.0. I'm on Windows 7 x64 SP1.
Cheers.

**Comment by TomWright** [comment]
Very sorry about that, I'll jump right on it.

**Comment by TomWright** [comment]
I've tested `dasel_windows_386.exe` on my 64bit windows 10 machine and it works as expected.

The only difference I can see is that it will have built using go v1.22 instead of v1.21. From a quick google search it looks as though this is not in-fact a bug with dasel, rather the go team have dropped support for Windows 7 in newer versions of go:
- https://github.com/golang/go/issues/52188
- https://github.com/golang/go/issues/57003
- https://github.com/golang/go/issues/57004

While I would love to facilitate as many users as possible, locking dasel to an old version of the language feels like a bit of a stretch to me.

While it's unfortunate you cannot execute the binary itself, I believe the following are your options:
1. Stay on `v2.5.0`. Assuming you don't need any of the new features/bug fixes then any current scripts/workflows you have should continue to work.
2. Use dasel via docker. You can always use the latest version of dasel without worrying about compatibility with older versions of Windows.
3. Build dasel yourself with an older version of go. I expect you'd need at least `v1.21`, but if that doesn't work it sounds like `v1.20` will.

Here's a simple example of running the latest version of dasel with Docker:
```
$ echo '{"foo":"bar"}' | docker run -i --rm ghcr.io/tomwright/dasel:2.6.0 -r json 'foo'
"bar"
```

If you have any questions or doubts please let me know.

**Comment by TomWright** [comment]
_As announced in the Go 1.20 release notes, Go 1.21 requires at least Windows 10 or Windows Server 2016; support for previous versions has been discontinued._
https://tip.golang.org/doc/go1.21

**Comment by hellishvictor** [comment]
And the thing is that two days ago I downloaded go v1.22 and goreleaser for compile (by very first time) a go program, "[micro](https://github.com/zyedidia/micro/issues/3136)". Now already got the [1.20.14](https://go.dev/dl/go1.20.14.windows-arm64.zip).
I'll try to compile it, but I must admit that v2.5.0 works like a charm. Definitely, you've created an amazing tool, man.
I haven't got into docks yet, but it's mentioned everywhere, so when I got time I'll give it a try.
Thank you very much!
